jobs/deploy-buildmaster.groovy

```diff
@@ -27,6 +27,7 @@ currentBuild.displayName = "${currentBuild.displayName} - ${params.GIT_BRANCH} -
 def buildAndDeploy() {
   withTimeout('15m') {
     kaGit.quickClone("git@github.com:Khan/buildmaster2", "buildmaster2", params.GIT_BRANCH);
+    kaGit.safePullInBranch("buildmaster2", params.GIT_BRANCH);
```

Comment:
I think you just want kaGit.safeSyncToOrigin.  There's no need to do anything fancy here, especially with such a small repo.

Reply:
> When we initially tried to use safeSyncToOrigin, the git repo couldn't be found. It seemed to us like safeSyncToOrigin was specifically meant to be used for the webapp repo, is that not the case?

That is not the case.

What is the actual error you got?

> Doing a clone and pull seems like the simplest option.

It's the simplest, but not the most correct.  What happens if somehow the workspace gets dirty?  What if we add submodules?  There are a million things that can go wrong, which we have hardened our scripts to over time.  safeSyncToOrigin is the reliable way to get a particular repo to a particular commit.  As the comments say in kaGit.groovy:
```
// Clone a repo, using our existing workdirs if possible.
// This is very low-level, and does not update submodules!  You probably want
// safeSyncToOrigin, below, instead.
def quickClone(repoToClone, directory, commit) {
   exec(["jenkins-jobs/safe_git.sh", "clone", repoToClone, directory, commit]);
}
```